oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -73,8 +75,46 @@ abstract static class CredentialSource {
     }
   }
 
+  /**
+   * Encapsulates the service account impersonation options portion of the configuration for
+   * ExternalAccountCredentials.
+   *
+   * <p>If token_lifetime_seconds is not specified, the library will default to a 1-hour lifetime.
+   *
+   * <pre>
+   * Sample credential source for Pluggable Auth credentials:
```

Comment:
This actually isn't in the credential source. Also should just be generic to External Credentials